Updates to support Fedora

spoore1 opened the pull request:
controller.py:
- dnf groups 'Server with GUI' and 'Smart Card Support' do not exist in Fedora.  Running dnf installs for minimal packages in these cases.
- python3-uinput is needed in newer version of Fedora instead of the older python-uinput used from Pypi.  Older version uses a deprecated sysconfig variable (SO) which is fixed in python3-uinput.
- rsyslog is needed by the tests for checking /var/log/secure.

setup.py:
- dropping python-uinput to let the controller install the appropriate version.

Jakuje commented on SCAutolib/controller.py:
```diff
@@ -184,8 +188,11 @@ def setup_system(self, install_missing: bool, gdm: bool, graphical: bool):
+            run(['dnf', 'groupinstall', "Smart Card Support", '-y'])
+            logger.debug("Smart Card Support group in installed.")
+        else:
+            run(['dnf', 'install', 'esc', 'pcsc-lite-ccid', '-y'])
```
I would put here also `opensc` and maybe remove `esc` as I do not think this is used for anything useful in Fedora (I think it does not even work now).

Jakuje commented:
Technically looks good to me. Again, I would wait for George to make sure this works, but I do not expect any hiccups.

One note is the linter warning that might make sense to address:
 
> SCAutolib/models/gui.py:397:43: W292 no newline at end of file

spoore1 commented:
> Technically looks good to me. Again, I would wait for George to make sure this works, but I do not expect any hiccups.
> 
> One note is the linter warning that might make sense to address:
> 
> > SCAutolib/models/gui.py:397:43: W292 no newline at end of file

@Jakuje thanks for the review.

I think I fixed the linter issue and I added the check_home_screen function.   Was that in place when you approved?   or do you want to take another look at that?

GeorgePantelakis commented:
@spoore1 looks good to me too. I will test it shortly. 
Out of curiosity why did you use `check_home_screen(self, *args)` and check the `len` of args afterward to have default value and not just use `check_home_screen(self, polarity=True)`  instead? Also, could you please add a Docstring to the function? 
Those are nits. Then if you verify that it is the final code we can merge! Thank you very much for your contribution!

GeorgePantelakis commented on SCAutolib/models/gui.py:
```diff
@@ -385,3 +387,24 @@ def assert_no_text(self, key: str, timeout: float = 0):
                                 f"in the screenshot {screenshot}")
+
+    @log_decorator
+    def check_home_screen(self, *args):
```
Could you please add a Docstring to the function?

spoore1 replied:
done

GeorgePantelakis commented on SCAutolib/models/gui.py:
```diff
@@ -385,3 +387,24 @@ def assert_no_text(self, key: str, timeout: float = 0):
+
+    @log_decorator
+    def check_home_screen(self, *args):
+        if len(args) == 0:
```
nit: Out of curiosity why did you use check_home_screen(self, *args) and check the len of args afterward to have default value and not just use check_home_screen(self, polarity=True) instead?

spoore1 replied:
I tried initially but, hit an issue which I am not seeing now.  I'm guessing I had a typo somewhere that I missed but, I'm not sure.

I've switched to use check_home_screen(self, polarity=True)  since that reduces the necessary code.

GeorgePantelakis commented on setup.py:
```diff
@@ -25,7 +24,7 @@
 setup(
     name="SCAutolib",
-    version="3.1.2",
+    version="3.1.3",
```
nit: Since this is kind of a big change, in the sense that from now on Fedora will be supported, can you please change the version to 3.2.0 instead of 3.1.3?

spoore1 replied:
changed to 3.2.0.

GeorgePantelakis commented on SCAutolib/controller.py:
```diff
@@ -197,6 +206,13 @@ def setup_system(self, install_missing: bool, gdm: bool, graphical: bool):
 
+        # Fedora requires python3-uinput from RPM and rsyslog
+        if os_version == OSVersion.Fedora:
+            run(['dnf', 'install', 'python3-uinput', 'rsyslog', '-y'])
```
I tried to run the test in a fresh fedora VM and I found out that packages "openssl" and "nss-tools" (which are needed) were not installed. So could you please add them? Perhaps not here but in line 147 on controller where the general packages variable is defined.

spoore1 replied:
I am adding openssl and nss-tools to the controller location described.

I do see those packages listed in requirements.txt so, should I add those pip packages to a new section to install specific pip packages for graphical?

GeorgePantelakis replied:
Oh yeah, my bad. For some reason, they weren't installed when I ran the tests and I completely missed that they are there. Perhaps I did something wrong and did not install SCAutolib[graphical]. Let's consider it resolved.

GeorgePantelakis reviewed (commented):
Just some minor changes. Mostly nitpicking staff. In general, it looks good to me too.

GeorgePantelakis commented:
I have tested the changes (+ the nitpicks) on Fedora 39 and Fedora 38 and the tests are passing. I haven't tested all the tests but the most important ones are ok. If some tweaks are needed we can do them later if necessary.

GeorgePantelakis commented:
Run the tests again and they pass on f39 and f38 with the changes (except for the lock_on_removal one which is expected after the mail discussion). I think we can merge @spoore1 if you are finished.

GeorgePantelakis commented:
Hello @spoore1, have you done with this PR? I remember you said that you wanted to add some more code but I do not see any movement so perhaps I was mistaken.

GeorgePantelakis commented:
Hello @spoore1,

Since this PR is considered done and intertwined with the SC-tests PR, I will go ahead and merge both of them. There is no new code added and the old code has been tested already. Feel free to create a new PR for both projects as needed. Thank you very much for your contribution!
